herd/AArch64Sem.ml

```diff
@@ -1545,15 +1554,17 @@ module Make
           else if checked then
             lift_memtag_virt mop ma dir an ii
           else
+            let ma = if C.mte_store_only then ma >>= fun a -> loc_extract a else ma in
```

Comment:
I am a bit dubious about this but I am also wondering if this is better done inside `do_ldr` and `cas`.